src/commands/createFunctionApp/stacks/FunctionAppStackStep.ts

```diff
@@ -59,12 +40,17 @@ export class FunctionAppStackStep extends AzureWizardPromptStep<IFunctionAppWiza
     }
 
     public async getSubWizard(context: IFunctionAppWizardContext): Promise<IWizardOptions<IFunctionAppWizardContext> | undefined> {
+        const promptSteps: AzureWizardPromptStep<IFunctionAppWizardContext>[] = [];
+        if (shouldShowEolWarning(context.newSiteStack?.minorVersion)) {
+            promptSteps.push(new FunctionAppEOLWarningStep());
+        }
         if (context.newSiteOS === undefined) {
-            return { promptSteps: [new SiteOSStep()] };
+            promptSteps.push(new SiteOSStep())
         } else {
             await setLocationsTask(context);
             return undefined;
```

Comment:
The warning step isn't showing up for me since we're returning undefined here.

Reply:
Hmm that's weird it seems to show up for me.

Reply:
If `context.newSiteOS` is defined it will enter the else block and always return `undefined`. We need to make sure we always return the EOL step if necessary.